ntoskrnl/mm/ARM3/expool.c

```diff
@@ -2081,7 +2094,11 @@ ExAllocatePoolWithTag(IN POOL_TYPE PoolType,
             Entry->PoolTag = Tag;
             (POOL_FREE_BLOCK(Entry))->Flink = NULL;
             (POOL_FREE_BLOCK(Entry))->Blink = NULL;
-            return POOL_FREE_BLOCK(Entry);
+            Allocation = POOL_FREE_BLOCK(Entry);
+#if DBG
+            RtlFillMemory(Allocation, NumberOfBytes, 0xCD);
```

Comment:
It seems like this could be quite the perf hit. I wonder if it should be guarded by more than `DBG`...

Reply:
We can run this PR on the testbots to see whether it's indeed the case.

Reply:
No notable perf impact. See https://reactos.org/testman/compare.php?ids=68947,68949,68952,68953